services/QuillLMS/app/controllers/snapshots_controller.rb

```diff
@@ -43,12 +43,26 @@ def data_export
   end
 
   def options
-    render json: {
+    render json: build_options_hash
+  end
+
+  private def build_options_hash
+    {
       timeframes: Snapshots::Timeframes.frontend_options,
       schools: format_option_list(school_options),
       grades: GRADE_OPTIONS,
       teachers: format_option_list(sorted_teacher_options),
       classrooms: format_option_list(classroom_options)
+    }.merge(initial_load_options)
```

Comment:
I quite like the way that you've implemented this `initial_load?` override to keep the code simple and readable.